packages/client/hmi-client/src/services/notificationEventHandlers.ts

```diff
@@ -141,17 +139,48 @@ export const createNotificationEventHandlers = (notificationItems: Ref<Notificat
 	// Register handlers for each client event type
 
 	registerHandler<ExtractionStatusUpdate>(ClientEventType.ExtractionPdf, (event, created) => {
-		created.assetId = event.data.documentId;
-		getDocumentAsset(event.data.documentId).then((document) =>
-			Object.assign(created, { assetName: document?.name || '' })
+		created.assetId = event.data.data.documentId;
+		created.pageType = AssetType.Document;
+		getDocumentAsset(created.assetId, created.projectId).then((document) =>
+			Object.assign(created, { sourceName: document?.name || '' })
 		);
 	});
-
 	registerHandler<TaskResponse>(ClientEventType.TaskGollmModelCard, (event, created) => {
 		created.supportCancel = true;
-		created.assetId = event.data.additionalProperties.documentId;
-		getDocumentAsset(created.assetId).then((document) =>
-			Object.assign(created, { assetName: document?.name || '' })
+		created.assetId = event.data.additionalProperties.documentId as string;
+		created.pageType = AssetType.Document;
+		getDocumentAsset(created.assetId, created.projectId).then((document) =>
+			Object.assign(created, { sourceName: document?.name || '' })
+		);
+	});
+	registerHandler<TaskResponse>(ClientEventType.TaskGollmConfigureModel, (event, created) => {
+		created.supportCancel = true;
+		created.sourceName = 'Configure model';
+		created.assetId = event.data.additionalProperties.workflowId as string;
+		created.pageType = AssetType.Workflow;
+		created.nodeId = event.data.additionalProperties.nodeId as string;
+		getWorkflow(created.assetId, created.projectId).then((workflow) =>
+			Object.assign(created, { context: workflow?.name || '' })
+		);
```

Comment:
I don't understand why we are grabbing workflow here, or how `created.assetId` is a valid workflow id??

Reply:
When a source link is clicked in the each notification item, we are navigating user to the right asset or workflow (and operator modal) page with the corresponding project. WorkflowId is used for that purpose. The ids are passed and carried along with the request when the request is sent by the user from the specific context.

Reply:
So if all you need i workflow-name, why isn't that passed into the original request? It feels like we splitting something that should be contiguous into multiple steps across multiple components.

Reply:
Actually `workflowId` is needed to get the correct page url and the workflow name is for display purpose. We could pass the name into the original request but as I mentioned in other comments, at first It seemed it wasn't very clear to have an arbitrary object as a parameter value and I didn't want to add more parameters to the api endpoints. I think that's something we can improve as we go though.